server/db/driver/pg/penalties.go

```diff
@@ -0,0 +1,77 @@
+// This code is available on the terms of the project LICENSE.md file,
+// also available online at https://blueoakcouncil.org/license/1.0.0.
+
+package pg
+
+import (
+	"database/sql"
+	"fmt"
+	"time"
+
+	"decred.org/dcrdex/dex/encode"
+	"decred.org/dcrdex/server/account"
+	"decred.org/dcrdex/server/db"
+	"decred.org/dcrdex/server/db/driver/pg/internal"
+)
+
+// InsertPenalty adds penalty to the penalties table. The passed ID and Forgiven
+// fields are ignored. The db decided id is returned.
+func (a *Archiver) InsertPenalty(penalty *db.Penalty) (id int64) {
+	stmt := fmt.Sprintf(internal.InsertPenalty, penaltiesTableName)
+	a.db.QueryRow(stmt, penalty.AccountID, penalty.BrokenRule,
```

Comment:
error dropped.  I might be OK with that given a log and a comment, mainly because there's nothing we can do about it and I'm not eager to put the DB backend in a fatal state over this.

Reply:
I did not mean to ignore the error. Is it ok to just a.fatalBackendErr(err) or does more need to happen?

Reply:
Since the caller of `InsertPenalty` can't do much with the error, and I'd prefer not to throw the DB into a fatal state over this, let's just log the err.

I'll give this PR some attention now, sorry for the delay.

But I'd like to focus on solving the most pressing issue around penalization and account suspension, which is the practical issue of "what do I do now that my account is suspended and cannot place new orders?"  Basically the account needs to be able to finish active swaps, and when they are done, do the reinstatement however that turns out to be.  Penalization has only been blocking new orders, but should unbook all existing orders for the user to prevent further swaps from beginning.  Then perhaps after all active swaps are done, disconnect the user to force the re-registration.

Also, the penalties table in this PR is needed to provide penalization state on `'connect'` but I believe notifications should be sent when a violation results in a penalization (trade suspension).

Reply:
I decided to return an error if that's alright.